bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -321,10 +312,10 @@ public Module getModule(String moduleId) {
      * @return list of modules of defined type or all modules when the type is not
      *         specified.
      */
-    @SuppressWarnings("unchecked")
+    @SuppressWarnings({ "unchecked", "null" })
```

Comment:
Don't initialize result with null, but use `final List<T> result;` first.
Then add an else branch
```Java
else {
    result = Collections.emptyList();
}
```

you could then use a simple `return result;` at the end.

Or miss I something?
Why do you need to suppress the "null" warning?

Reply:
OK